packages/freezed/lib/src/templates/abstract_template.dart

```diff
@@ -49,27 +51,26 @@ ${copyWith.commonContreteImpl(abstractProperties)}
 
   String get _toJson {
     if (!shouldGenerateJson) return '';
-
-    return 'Map<String, dynamic> toJson();';
+    return 'Map<String, dynamic> toJson() => throw $privConstUsedErrorVarName;';
   }
 
   String get _when {
     if (!allConstructors.shouldGenerateUnions) return '';
-    return '${whenPrototype(allConstructors)};';
+    return '${whenPrototype(allConstructors)} => throw $privConstUsedErrorVarName;';
   }
 
   String get _maybeWhen {
     if (!allConstructors.shouldGenerateUnions) return '';
-    return '${maybeWhenPrototype(allConstructors)};';
+    return '${maybeWhenPrototype(allConstructors)} => throw $privConstUsedErrorVarName;';
   }
 
   String get _map {
     if (!allConstructors.shouldGenerateUnions) return '';
-    return '${mapPrototype(allConstructors, genericsParameter)};';
+    return '${mapPrototype(allConstructors, genericsParameter)} => throw $privConstUsedErrorVarName;';
   }
 
   String get _maybeMap {
     if (!allConstructors.shouldGenerateUnions) return '';
-    return '${maybeMapPrototype(allConstructors, genericsParameter)};';
+    return '${maybeMapPrototype(allConstructors, genericsParameter)} => throw $privConstUsedErrorVarName;';
```

Comment:
we need tests for all of those exceptions.

Reply:
True, doing it tonight also.